diff -Naur xkeyboard-config-1.3orig/debian/changelog xkeyboard-config-1.3bepo1.0/debian/changelog
--- xkeyboard-config-1.3orig/debian/changelog	2008-09-19 21:08:58.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/debian/changelog	2009-06-17 07:33:58.000000000 +0200
@@ -1,3 +1,18 @@
+xkeyboard-config (1.3-2.1bepo.1.0+tm2) unstable; urgency=low
+
+  * met à jour la description géométrique du clavier Typematrix
+    TM2030USB
+
+ -- Frédéric Boiteux <fboiteux@free.fr>  Wed, 17 Jun 2009 07:33:58 +0200
+
+xkeyboard-config (1.3-2.1bepo.1.0) unstable; urgency=low
+
+  * installe la disposition Dvorak française « Bépo » en version 1.0, et
+    nettoie les versions de développement.
+  * Ajoute la description du clavier TypeMatrix TM2030USB.
+
+ -- Frédéric Boiteux <fboiteux@free.fr>  Wed, 03 Jun 2009 21:37:17 +0200
+
 xkeyboard-config (1.3-2) unstable; urgency=high
 
   * Fix compat rules for Romanian layout (closes: #485702).
diff -Naur xkeyboard-config-1.3orig/debian/geometry.dir xkeyboard-config-1.3bepo1.0/debian/geometry.dir
--- xkeyboard-config-1.3orig/debian/geometry.dir	2008-09-19 21:08:58.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/debian/geometry.dir	2009-06-03 21:33:11.000000000 +0200
@@ -93,6 +93,7 @@
 -------- -------- sun(type6unix)
 -------- -------- sun(x86)
 -------- -------- sun(x86tuv)
+-------- -------- typematrix(tm2030USB)
 -------- -------- thinkpad(common)
 -------- -------- thinkpad(intl)
 -------- -------- thinkpad(us)
diff -Naur xkeyboard-config-1.3orig/debian/symbols.dir xkeyboard-config-1.3bepo1.0/debian/symbols.dir
--- xkeyboard-config-1.3orig/debian/symbols.dir	2008-09-19 21:08:58.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/debian/symbols.dir	2009-06-03 21:40:09.000000000 +0200
@@ -165,6 +165,8 @@
 --p----- a------- fo(nodeadkeys)
 --p----- a------- fr(Sundeadkeys)
 -dp----- a------- fr(basic)
+--p----- a------- fr(bepo_1_0)
+--p----- a------- fr(bepo_1_0_latin9)
 --p----- a------- fr(bepo)
 --p----- a------- fr(bepo_latin9)
 --p----- a------- fr(dvorak)
diff -Naur xkeyboard-config-1.3orig/geometry/geometry.dir xkeyboard-config-1.3bepo1.0/geometry/geometry.dir
--- xkeyboard-config-1.3orig/geometry/geometry.dir	2007-05-31 23:17:46.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/geometry/geometry.dir	2009-06-03 21:50:51.000000000 +0200
@@ -83,3 +83,4 @@
 -d------ -------- keytronic(FlexPro)
 -d------ -------- kinesis(model100)
 -d------ -------- macintosh(macintosh)
+-------- -------- typematrix(tm2030USB)
diff -Naur xkeyboard-config-1.3orig/geometry/Makefile.am xkeyboard-config-1.3bepo1.0/geometry/Makefile.am
--- xkeyboard-config-1.3orig/geometry/Makefile.am	2007-07-12 00:42:44.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/geometry/Makefile.am	2009-06-04 18:20:11.000000000 +0200
@@ -8,6 +8,6 @@
 hhk hp keytronic kinesis \
 macintosh microsoft nec \
 northgate pc sony thinkpad \
-sun winbook README
+sun winbook typematrix README
 
 include $(top_srcdir)/xkbrules.am
diff -Naur xkeyboard-config-1.3orig/geometry/Makefile.in xkeyboard-config-1.3bepo1.0/geometry/Makefile.in
--- xkeyboard-config-1.3orig/geometry/Makefile.in	2008-05-28 01:11:52.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/geometry/Makefile.in	2009-06-04 18:25:09.000000000 +0200
@@ -210,7 +210,7 @@
 hhk hp keytronic kinesis \
 macintosh microsoft nec \
 northgate pc sony thinkpad \
-sun winbook README
+sun winbook typematrix README
 
 dist_dir_DATA = $(subdir).dir
 dirdir = $(xkb_base)
diff -Naur xkeyboard-config-1.3orig/geometry/typematrix xkeyboard-config-1.3bepo1.0/geometry/typematrix
--- xkeyboard-config-1.3orig/geometry/typematrix	1970-01-01 01:00:00.000000000 +0100
+++ xkeyboard-config-1.3bepo1.0/geometry/typematrix	2009-06-17 07:55:24.000000000 +0200
@@ -0,0 +1,180 @@
+// -*- indent-tabs-mode: nil -*-
+
+// Created by Frédéric Boiteux <fboiteux (at) free (dot) fr>
+
+xkb_geometry "tm2030USB" {
+    description = "Typematrix 2030 USB";
+
+    // Note : les touches spéciales (Calc, www, desktop...) sont nommées ici
+    // <I[XX]>, pour qu elles soient correctement affichées par l'applet Gnome.
+    // Les numéros XX sont inventés, en fait, elles ne renvoient pas
+    // de scancode (Fn, touches multimédia)
+    // ou plusieurs scancodes (comme la touche Desktop, Shuffle…).
+
+    // Taille globale du clavier :
+    width  = 320;
+    height = 130;
+    // couleurs de fond et de dessin :
+    baseColor  = "black";
+    labelColor = "black";
+
+    // formes qui seront reprises ensuite :
+    // touche de base : 20mm de coté
+    shape "NORM" { cornerRadius=1, { [18, 18]   }, { [1.5, 1.5], [16.5, 14] } };
+    shape "FUNC" { cornerRadius=1, { [18, 12]   }, { [2.5,   0], [15.5, 10] } };
+    shape "LFSH" { cornerRadius=1, { [18, 36.5] }, { [2.5,   0], [15.5, 30] } };
+    shape "ALT1" { cornerRadius=1, { [27.5, 18] }, { [2.5,   0], [23.5, 14] } };
+    shape "BKSP" { cornerRadius=1, { [18, 30.5] }, { [1.5, 1.5], [16.5, 29] } };
+    shape "RTSN" { cornerRadius=1, { [18, 37.5] }, { [1.5, 1.5], [16.5, 35] } };
+    shape "SPCE" { cornerRadius=1, { [92, 25]   }, { [1.5, 1.5], [90.5, 22] } };
+
+    // Rangée des touches de fonction
+    section "Function" {
+        left = 27;
+        top  = 1.5;
+        key.gap = 0.5;
+        key.color = "grey50";
+        key.shape = "FUNC";
+        row {
+            keys { <ESC>,  <FK01>, <FK02>, <FK03>, <FK04>, <FK05> };
+        };
+        row {
+            left = 129.5;
+            keys { <FK06>, <FK07>, <FK08>, <FK09>, <FK10>, <FK11>,
+                   <FK12>, <NMLK> };
+        };
+    };
+
+    // Partie alpha-numérique du clavier
+    section "Alpha" {
+        left = 27;
+        top  = 14;
+        key.gap = 0.5;
+        key.color = "grey10";
+        key.shape = "NORM";
+        row {
+            top  = 0;
+            keys { <TLDE>,             <AE01>, <AE02>, <AE03>, <AE04>, <AE05> };
+        };
+        row {
+            top  = 18.5;
+            keys { <TAB>,              <AD01>, <AD02>, <AD03>, <AD04>, <AD05> };
+        };
+        row {
+            top  = 37;
+            keys { { <LFSH>, "LFSH" }, <AC01>, <AC02>, <AC03>, <AC04>, <AC05> };
+        };
+        row {
+            top  = 55.5;
+            left = 18.5;
+            keys {                     <AB01>, <AB02>, <AB03>, <AB04>, <AB05> };
+        };
+
+        row {
+            top  = 0;
+            left = 129.5;
+            // I01 pour la touche Calc
+            keys { <AE06>, <AE07>, <AE08>, <AE09>, <AE10>, <AE11>, <AE12>, <I01> };
+        };
+        row {
+            top  = 18.5;
+            left = 129.5;
+            // I02 pour la touche Mail
+            keys { <AD06>, <AD07>, <AD08>, <AD09>, <AD10>, <AD11>, <AD12>, <I02> };
+        };
+        row {
+            top  = 37;
+            left = 129.5;
+            keys { <AC06>, <AC07>, <AC08>, <AC09>, <AC10>, <AC11>, { <RTSH>, "LFSH" }, <CAPS> };
+        };
+        row {
+            top  = 55.5;
+            left = 129.5;
+            keys { <AB06>, <AB07>, <AB08>, <AB09>, <AB10>, <AB11> };
+        };
+        row {
+            top  = 55.5;
+            left = 259;
+            // I03 pour la touche WWW
+            keys { <I03> };
+        };
+
+        row {
+            top  = 74;
+            left = 74;
+            keys { { <SPCE>, "SPCE" } };
+        };
+    };
+
+    // touches de contrôle
+    section "Control" {
+        left = 27;
+        top  = 0;
+        key.gap = 0.5;
+        key.color = "grey30";
+        key.shape = "NORM";
+        row {
+            top  = 88;
+            // I04 pour la touche Play, I05 pour la touche Shuffle
+            keys { <LCTL>, <I04>, <MENU>, <I05> };
+        };
+        row {
+            top = 106.5;
+            // I06 pour la touche Fn
+            keys { <I06>, { <LWIN>, "ALT1" }, { <LALT>, "ALT1" } };
+        };
+
+        row {
+            top  = 88;
+            left = 166.5;
+            // I07 pour la touche Desktop
+            keys { <I07>, <HOME>, <UP>, <END>, { <RCTL>, "LFSH"}, <PGUP> };
+        };
+        row {
+            top = 106.5;
+            left = 166.5;
+            keys { <RALT>, <LEFT>, <DOWN>, <RGHT> };
+        };
+        row {
+            top = 106.5;
+            left = 259;
+            keys { <PGDN> };
+        };
+
+        row {
+            top  = 1;
+            left = 111.5;
+            vertical = true;
+            keys { <DELE>, { <BKSP>, "BKSP" }, { <RTRN>, "LFSH" } };
+        };
+    };
+
+    // autres décors
+    shape "FORME1" { cornerRadius=1, { [27, 126.5] } };
+    solid "panneau1" {
+        shape = "FORME1";
+        left  = 0;
+        top   = 0;
+        color = "grey40";
+    };
+
+    shape "FORME2" { cornerRadius=1, { [92, 13] } };
+    solid "panneau2" {
+        shape = "FORME2";
+        left = 101.5;
+        top = 113.5;
+        color = "grey40";
+    };
+
+    shape "LED"  { cornerRadius = 2, { [4, 4] } };
+
+    indicator.onColor  = "green";
+    indicator.offColor = "grey10";
+    indicator.left     = 10;
+    indicator.shape    = "LED";
+
+    indicator "Num Lock"     { top = 11; };
+    indicator "Caps Lock"    { top = 27.5; };
+    indicator "Scroll Lock"  { top = 44; };
+    indicator "Power"	     { top = 60.5; };
+};
diff -Naur xkeyboard-config-1.3orig/rules/base xkeyboard-config-1.3bepo1.0/rules/base
--- xkeyboard-config-1.3orig/rules/base	2008-05-10 01:22:14.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/rules/base	2009-06-03 21:43:31.000000000 +0200
@@ -142,6 +142,7 @@
   flexpro	=	keytronic(FlexPro)
   hp6000	=	hp(omnibook)
   omnikey101	=	northgate(omnikey101)
+  tm2030USB	=	typematrix(tm2030USB)
  $pcmodels	=	pc(%m)
  $pcgeometries	=	pc(%m)
   everex	=	everex(STEPnote)
diff -Naur xkeyboard-config-1.3orig/rules/base.lst xkeyboard-config-1.3bepo1.0/rules/base.lst
--- xkeyboard-config-1.3orig/rules/base.lst	2008-05-12 22:33:05.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/rules/base.lst	2009-06-03 21:47:20.000000000 +0200
@@ -83,6 +83,7 @@
   thinkpad        IBM ThinkPad 560Z/600/600E/A22E
   thinkpadintl    IBM ThinkPad 560Z/600/600E/A22E, Intl
   thinkpad60      IBM ThinkPad R60/T60/Z60m/Z60t/R61/T61/Z61m/Z61t
+  tm2030USB       Typematrix 2030 USB
   logiaccess      Logitech Access Keyboard
   logiclx300      Logitech Cordless Desktop LX-300
   logii350        Logitech Internet 350 Keyboard
@@ -366,6 +367,8 @@
   latin9          fr: (Legacy) Alternative
   latin9_nodeadkeys fr: (Legacy) Alternative, eliminate dead keys
   latin9_sundeadkeys fr: (Legacy) Alternative, Sun dead keys
+  bepo_1_0        fr: Bepo, ergonomic, Dvorak way, v1.0
+  bepo_1_0_latin9 fr: Bepo, ergonomic, Dvorak way, v1.0, latin-9 only
   bepo            fr: Bepo, ergonomic, Dvorak way
   bepo_latin9     fr: Bepo, ergonomic, Dvorak way, latin-9 only
   dvorak          fr: (Legacy) Dvorak
diff -Naur xkeyboard-config-1.3orig/rules/base.m_g.part xkeyboard-config-1.3bepo1.0/rules/base.m_g.part
--- xkeyboard-config-1.3orig/rules/base.m_g.part	2008-01-08 00:04:58.000000000 +0100
+++ xkeyboard-config-1.3bepo1.0/rules/base.m_g.part	2009-06-03 23:03:32.000000000 +0200
@@ -5,6 +5,7 @@
   flexpro	=	keytronic(FlexPro)
   hp6000	=	hp(omnibook)
   omnikey101	=	northgate(omnikey101)
+  tm2030USB	=	typematrix(tm2030USB)
  $pcmodels	=	pc(%m)
  $pcgeometries	=	pc(%m)
   everex	=	everex(STEPnote)
diff -Naur xkeyboard-config-1.3orig/rules/base.xml.in xkeyboard-config-1.3bepo1.0/rules/base.xml.in
--- xkeyboard-config-1.3orig/rules/base.xml.in	2008-05-12 22:33:00.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/rules/base.xml.in	2009-06-17 07:29:10.000000000 +0200
@@ -590,6 +590,13 @@
     </model>
     <model>
       <configItem>
+        <name>tm2030USB</name>
+        <description>Typematrix 2030 USB</description>
+        <vendor>Typematrix</vendor>
+      </configItem>
+    </model>
+    <model>
+      <configItem>
         <name>logiaccess</name>
         <_description>Logitech Access Keyboard</_description>
         <vendor>Logitech</vendor>
@@ -1976,7 +1983,7 @@
         <variant>
           <configItem>
             <name>dvorak</name>
-            <_description>Dvorak</_description>
+            <_description>(Legacy) Dvorak</_description>
           </configItem>
         </variant>
         <variant>
@@ -2197,6 +2204,18 @@
         </variant>
         <variant>
           <configItem>
+            <name>bepo_1_0</name>
+            <_description>Bepo, ergonomic, Dvorak way, v1.0</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>bepo_1_0_latin9</name>
+            <_description>Bepo, ergonomic, Dvorak way, v1.0, latin-9 only</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>bepo</name>
             <_description>Bepo, ergonomic, Dvorak way</_description>
           </configItem>
diff -Naur xkeyboard-config-1.3orig/symbols/fr xkeyboard-config-1.3bepo1.0/symbols/fr
--- xkeyboard-config-1.3orig/symbols/fr	2008-04-22 21:54:11.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/symbols/fr	2009-06-06 16:08:25.000000000 +0200
@@ -254,7 +254,7 @@
     key <AE03>	{ [         quotedbl,                3,           numbersign,                 breve ] }; // " 3 # ˘
 
     // Second row
-    key <AD11>	{ [  circumflex,   diaeresis,           asciitilde,                 Aring ] }; // ^ ̈ ̃ Å
+    key <AD11>	{ [  asciicircum,   diaeresis,           asciitilde,                 Aring ] }; // ^ ̈ ̃ Å
     key <AC11>	{ [           ugrave,          percent,                acute,                Ugrave ] }; // ù % ' Ù
     key <BKSL>	{ [         asterisk,               mu,                grave,                macron ] }; // * µ ` ̄
 };
@@ -430,98 +430,168 @@
 };
 
 // Bépo : Improved ergonomic french keymap using Dvorak method.
-// Built by community on 'Dvorak Fr' mailing list :
-// dvorak-fr@gpl.insa-lyon.fr and on its wiki : http://gpl.insa-lyon.fr/Dvorak-Fr/ .
-// XOrg integration in 2006 by Frédéric Boiteux <fboiteux @ free.fr>
+// Built by community, see : http://www.bepo.fr
+// XOrg integration (1.0rc2 version) in 2008 by Frédéric Boiteux
+// <fboiteux at free dot fr>
 //
-// Bépo layout (6.2.1.1 version) for a pc105 keyboard (french) :
+// Bépo layout (1.0 version) for a pc105 keyboard (french) :
 // ┌────┐
 // │ S A│   S = Shift,  A = AltGr + Shift
 // │ s a│   s = normal, a = AltGr
 // └────┘
 //
 // ┌─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┲━━━━━━━━━┓
-// │ $   │ 1 ¹ │ 2 ² │ 3 ³ │ 4 ° │ 5 ˚ │ 6   │ 7   │ 8   │ 9   │ 0   │ ^   │ µ   ┃ ⌫ Retour┃
-// │ @ # │ _ { │ " } │ « < │ » > │ ( [ │ ) ] │ + ± │ - ¬ │ * × │ / ÷ │ = ¯ │ %   ┃  arrière┃
+// │ # ¶ │ 1 „ │ 2 “ │ 3 ” │ 4 ≤ │ 5 ≥ │ 6   │ 7 ¬ │ 8 ¼ │ 9 ½ │ 0 ¾ │ ° ′ │ ` ″ ┃ ⌫ Retour┃
+// │ $ – │ " — │ « < │ » > │ ( [ │ ) ] │ @ ^ │ + ± │ - − │ / ÷ │ * × │ = ≠ │ % ‰ ┃  arrière┃
 // ┢━━━━━┷━┱───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┺━┳━━━━━━━┫
-// ┃       ┃ B   │ É ˊ │ P ¶ │ O Œ │ È ` │ Ç   │ V   │ D Ð │ L   │ F   │ J   │ Z   ┃Entrée ┃
-// ┃Tab ↹  ┃ b | │ é & │ p § │ o œ │ è ` │ ç   │ v   │ d ð │ l £ │ f ª │ j   │ z   ┃   ⏎   ┃
+// ┃       ┃ B ¦ │ É ˝ │ P § │ O Œ │ È ` │ !   │ V   │ D Ð │ L   │ J Ĳ │ Z Ə │ W   ┃Entrée ┃
+// ┃Tab ↹  ┃ b | │ é ˊ │ p & │ o œ │ è ` │ ˆ ¡ │ v ˇ │ d ð │ l / │ j ĳ │ z ə │ w ̆ ┃   ⏎   ┃
 // ┣━━━━━━━┻┱────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┺┓      ┃
-// ┃        ┃ A Æ │ U Ù │ I ˇ │ E   │ ?   │ C ¢ │ T Þ │ S   │ N   │ R   │ M   │ !   ┃      ┃
-// ┃Maj ⇬   ┃ a æ │ u ù │ i ̈ │ e € │ , ¿ │ c © │ t þ │ s ß │ n   │ r ® │ m º │ ˆ ¡ ┃      ┃
+// ┃        ┃ A Æ │ U Ù │ I ˙ │ E ¤ │ ; ̛ │ C ſ │ T Þ │ S ẞ │ R ™ │ N   │ M º │ Ç , ┃      ┃
+// ┃Maj ⇬   ┃ a æ │ u ù │ i ̈ │ e € │ , ’ │ c © │ t þ │ s ß │ r ® │ n ˜ │ m ¯ │ ç ¸ ┃      ┃
 // ┣━━━━━━━┳┹────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┲┷━━━━━┻━━━━━━┫
-// ┃       ┃ ù   │ À ˜ │ H   │ Y   │ : · │ K   │ ;   │ Q   │ G   │ X   │ W   ┃             ┃
-// ┃Shift ⇧┃ ê / │ à ~ │ h ! │ y ¥ │ . \ │ k   │ '   │ q   │ g   │ x   │ w   ┃Shift ⇧      ┃
+// ┃       ┃ Ê   │ À   │ Y ‘ │ X ’ │ : · │ K ⌨ │ ? ̉ │ Q ̣ │ G   │ H ‡ │ F ª ┃             ┃
+// ┃Shift ⇧┃ ê / │ à \ │ y { │ x } │ . … │ k ~ │ ' ¿ │ q ˚ │ g µ │ h † │ f ˛ ┃Shift ⇧      ┃
 // ┣━━━━━━━╋━━━━━┷━┳━━━┷━━━┱─┴─────┴─────┴─────┴─────┴─────┴───┲━┷━━━━━╈━━━━━┻━┳━━━━━━━┳━━━┛
-// ┃       ┃       ┃       ┃ ⍽ Espace insécable              ␣ ┃       ┃       ┃       ┃
-// ┃Ctrl   ┃Meta   ┃Alt    ┃ ␣ Espace                        ␣ ┃AltGr ⇮┃Menu   ┃Ctrl   ┃
+// ┃       ┃       ┃       ┃ Espace inséc.   Espace inséc. fin ┃       ┃       ┃       ┃
+// ┃Ctrl   ┃Meta   ┃Alt    ┃ ␣ (Espace)      _               ␣ ┃AltGr ⇮┃Menu   ┃Ctrl   ┃
 // ┗━━━━━━━┻━━━━━━━┻━━━━━━━┹───────────────────────────────────┺━━━━━━━┻━━━━━━━┻━━━━━━━┛
-partial alphanumeric_keys
-xkb_symbols "bepo" {
 
+
+// Version 1.0 :
+partial alphanumeric_keys
+xkb_symbols "bepo_1_0" {
+    include "level3(ralt_switch)"
     include "keypad(oss)"
 
-    name[Group1]= "France - Bepo, ergonomic, Dvorak way";
+    name[Group1]= "France - Bepo, ergonomic, Dvorak way, v1.0";
+
+    // Modifications depuis la version précédente :
+    // * Ajout de U2328 (⌨) en AltGr+Maj+K
+    // * remplace les codes UFDDx des touches mortes spécifiques par leurs
+    //   valeur « officielles » dans Xorg depuis leur intégration :
+    //   UFDD0 = dead_stroke     -> Ufe63
+    //   UFDD1 = dead_currency   -> Ufe6f
+    //   UFDD3 = dead_belowcomma -> Ufe6e
+    //   (UFDD2 = dead_greek reste en place en attendant son
+    //    intégration dans Xorg :-)
 
     // First row
-    key <TLDE> { [              at,       dollar,     numbersign                 ] }; // @ $ #
-    key <AE01> { [      underscore,            1,      braceleft,    onesuperior ] }; // _ 1 { ¹
-    key <AE02> { [        quotedbl,            2,     braceright,    twosuperior ] }; // " 2 } ²
-    key <AE03> { [   guillemotleft,            3,           less,  threesuperior ] }; // « 3 < ³
-    key <AE04> { [  guillemotright,            4,        greater,         degree ] }; // » 4 > °
-    key <AE05> { [       parenleft,            5,    bracketleft, dead_abovering ] }; // ( 5 [ ˚
-    key <AE06> { [      parenright,            6,   bracketright                 ] }; // ) 6 ]
-    key <AE07> { [            plus,            7,      plusminus                 ] }; // + 7 ±
-    key <AE08> { [           minus,            8,        notsign                 ] }; // - 8 ¬
-    key <AE09> { [        asterisk,            9,       multiply                 ] }; // * 9 ×
-    key <AE10> { [           slash,            0,       division                 ] }; // / 0 ÷
-    key <AE11> { [           equal,  asciicircum,         macron                 ] }; // = ^ ¯
-    key <AE12> { [         percent,           mu                                 ] }; // % µ
+    key <TLDE> { [          dollar,   numbersign,        endash,       paragraph ] }; // $ # – ¶
+    key <AE01> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [        quotedbl,            1,         emdash, doublelowquotemark ] }; // " 1 — „
+    key <AE02> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [   guillemotleft,            2,           less,  leftdoublequotemark ] }; // « 2 < “
+    key <AE03> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [  guillemotright,            3,        greater, rightdoublequotemark ] }; // » 3 > ”
+    key <AE04> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [       parenleft,            4,    bracketleft,      lessthanequal ] }; // ( 4 [ ≤
+    key <AE05> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [      parenright,            5,   bracketright,   greaterthanequal ] }; // ) 5 ] ≥
+    key <AE06> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [              at,            6,    asciicircum                 ] }; // @ 6 ^
+    key <AE07> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [            plus,            7,      plusminus,        notsign ] }; // + 7 ± ¬
+    key <AE08> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [           minus,            8,          U2212,     onequarter ] }; // - 8 − ¼
+    key <AE09> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [           slash,            9,       division,        onehalf ] }; // / 9 ÷ ½
+    key <AE10> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [        asterisk,            0,       multiply,  threequarters ] }; // * 0 × ¾
+    key <AE11> { [           equal,       degree,       notequal,        minutes ] }; // = ° ≠ ′
+    key <AE12> { [         percent,        grave,          U2030,        seconds ] }; // % ` ‰ ″
 
     // Second row
-    key <AD01> { [               b,            B,            bar                 ] }; // b B |
-    key <AD02> { [          eacute,       Eacute,      ampersand,     dead_acute ] }; // é É & ˊ
-    key <AD03> { [               p,            P,        section,      paragraph ] }; // p P § ¶
+    key <AD01> { [               b,            B,            bar,      brokenbar ] }; // b B | ¦
+    key <AD02> { [          eacute,       Eacute,     dead_acute, dead_doubleacute ] }; // é É ˊ ˝
+    key <AD03> { [               p,            P,      ampersand,        section ] }; // p P & §
     key <AD04> { [               o,            O,             oe,             OE ] }; // o O œ Œ
-    key <AD05> { [          egrave,       Egrave,          grave,     dead_grave ] }; // è È ` `
-    key <AD06> { [        ccedilla,     Ccedilla                                 ] }; // ç Ç
-    key <AD07> { [               v,            V                                 ] }; // v V
+    key <AD05> { [          egrave,       Egrave,     dead_grave,          grave ] }; // è È ` `
+    key <AD06> { [ dead_circumflex,       exclam,     exclamdown                 ] }; // ^ ! ¡
+    key <AD07> { [               v,            V,     dead_caron                 ] }; // v V ˇ
     key <AD08> { [               d,            D,            eth,            ETH ] }; // d D ð Ð
-    key <AD09> { [               l,            L,       sterling                 ] }; // l L £
-    key <AD10> { [               f,            F,    ordfeminine                 ] }; // f F ª
-    key <AD11> { [               j,            J                                 ] }; // j J
-    key <AD12> { [               z,            Z                                 ] }; // z Z
+    key <AD09> { [               l,            L,          Ufe63                 ] }; // l L /
+    key <AD10> { [               j,            J,          U0133,          U0132 ] }; // j J ĳ Ĳ
+    key <AD11> { [               z,            Z,          schwa,          SCHWA ] }; // z Z ə Ə
+    key <AD12> { [               w,            W,     dead_breve                 ] }; // w W ̆
 
     // Third row
     key <AC01> { [               a,            A,             ae,             AE ] }; // a A æ Æ
     key <AC02> { [               u,            U,         ugrave,         Ugrave ] }; // u U ù Ù
-    key <AC03> { [               i,            I, dead_diaeresis,     dead_caron ] }; // i I ̈ ˇ
-    key <AC04> { [               e,            E,       EuroSign                 ] }; // e E €
-    key <AC05> { [           comma,     question,   questiondown                 ] }; // , ? ¿
-    key <AC06> { [               c,            C,      copyright,           cent ] }; // c C © ¢
-    key <AC07> { [               t,            T,          thorn,          Thorn ] }; // t T þ Þ
-    key <AC08> { [               s,            S,         ssharp                 ] }; // s S ß
-    key <AC09> { [               n,            N                                 ] }; // n N
-    key <AC10> { [               r,            R,     registered                 ] }; // r R ®
-    key <AC11> { [               m,            M,      masculine                 ] }; // m M º
-    key <BKSL> { [ dead_circumflex,       exclam,     exclamdown                 ] }; // ^ ! ¡
+    key <AC03> { [               i,            I, dead_diaeresis,  dead_abovedot ] }; // i I ̈ ˙
+    key <AC04> { [               e,            E,       EuroSign,          Ufe6f ] }; // e E € ¤
+    key <AC05> { [           comma,    semicolon, rightsinglequotemark, dead_horn ] }; // , ; ’ ̛
+    key <AC06> { [               c,            C,      copyright,          U017F ] }; // c C © ſ
+    key <AC07> { [               t,            T,          thorn,          THORN ] }; // t T þ Þ
+    key <AC08> { [               s,            S,         ssharp,          U1E9E ] }; // s S ß ẞ
+    key <AC09> { [               r,            R,     registered,      trademark ] }; // r R ® ™
+    key <AC10> { [               n,            N,     dead_tilde                 ] }; // n N ~
+    key <AC11> { [               m,            M,    dead_macron,      masculine ] }; // m M ̄ º
+    key <BKSL> { [        ccedilla,     Ccedilla,   dead_cedilla,          Ufe6e ] }; // ç Ç ¸ ,
 
     // Fourth row
-    key <LSGT> { [     ecircumflex,       ugrave,          slash                 ] }; // ê ù /
-    key <AB01> { [          agrave,       Agrave,     asciitilde,     dead_tilde ] }; // à À ~ ~
-    key <AB02> { [               h,            H,         exclam                 ] }; // h H !
-    key <AB03> { [               y,            Y,            yen                 ] }; // y Y ¥
-    key <AB04> { [          period,        colon,      backslash, periodcentered ] }; // . : \ ·
-    key <AB05> { [               k,            K                                 ] }; // k K
-    key <AB06> { [      apostrophe,    semicolon                                 ] }; // ' ;
-    key <AB07> { [               q,            Q,         oslash,       Ooblique ] }; // q Q ø Ø
-    key <AB08> { [               g,            G                                 ] }; // g G
-    key <AB09> { [               x,            X                                 ] }; // x X
-    key <AB10> { [               w,            W                                 ] }; // w W
+    key <LSGT> { [     ecircumflex,  Ecircumflex,          slash                 ] }; // ê Ê /
+    key <AB01> { [          agrave,       Agrave,      backslash                 ] }; // à À \
+    key <AB02> { [               y,            Y,      braceleft, leftsinglequotemark  ] }; // y Y { ‘
+    key <AB03> { [               x,            X,     braceright, rightsinglequotemark ] }; // x X } ’
+    key <AB04> { [          period,        colon,       ellipsis, periodcentered ] }; // . : … ·
+    key <AB05> { [               k,            K,     asciitilde,          U2328 ] }; // k K ~ ⌨
+    key <AB06> { [      apostrophe,     question,   questiondown,      dead_hook ] }; // ' ? ¿ ̉
+    key <AB07> { [               q,            Q, dead_abovering,  dead_belowdot ] }; // q Q ˚ ̣
+    key <AB08> { [               g,            G,          UFDD2                 ] }; // g G µ
+    key <AB09> { [               h,            H,         dagger,   doubledagger ] }; // h H † ‡
+    key <AB10> { [               f,            F,    dead_ogonek,    ordfeminine ] }; // f F ̨ ª
 
-    key <SPCE> { [           space, nobreakspace,          space,          space ] }; // ␣ ⍽ (espace insécable) ␣ ␣
+    key <SPCE> { [           space, nobreakspace,     underscore,          U202F ] }; // ␣ (espace insécable) _ (espace insécable fin)
+};
 
-    include "level3(ralt_switch)"
+partial alphanumeric_keys
+xkb_symbols "bepo_1_0_latin9" {
+
+    // Restricts the fr(bepo_1_0) layout to latin9 symbols
+
+    include "fr(bepo_1_0)"
+    include "keypad(oss_latin9)"
+
+    name[Group1]="France - Bepo, ergonomic, Dvorak way, v1.0, latin-9 only";
+
+    key <TLDE> { [          dollar,   numbersign,        dollar,       paragraph ] }; // $ # $ ¶
+
+    key <AE01> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [        quotedbl,            1                                 ] }; // " 1
+    key <AE02> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [   guillemotleft,            2,           less                 ] }; // « 2 <
+    key <AE03> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [  guillemotright,            3,        greater                 ] }; // » 3 >
+    key <AE04> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [       parenleft,            4,    bracketleft                 ] }; // ( 4 [
+    key <AE05> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [      parenright,            5,   bracketright                 ] }; // ) 5 ]
+    key <AE08> { type[group1] = "FOUR_LEVEL_SEMIALPHABETIC", [           minus,            8,          minus,     onequarter ] }; // - 8 - ¼
+    key <AE11> { [           equal,       degree                                 ] }; // = °
+    key <AE12> { [         percent,        grave                                 ] }; // % `
+
+    key <AD01> { [               b,            B,            bar                 ] }; // b B |
+    key <AD02> { [          eacute,       Eacute,     dead_acute                 ] }; // é É ˊ
+    key <AD07> { [               v,            V                                 ] }; // v V
+    key <AD10> { [               j,            J                                 ] }; // j J
+    key <AD11> { [               z,            Z                                 ] }; // z Z
+    key <AD12> { [               w,            W                                 ] }; // w W
+
+    key <AC03> { [               i,            I, dead_diaeresis                 ] }; // i I ̈
+    key <AC05> { [           comma,    semicolon,          comma,      dead_horn ] }; // , ; , ̛
+    key <AC06> { [               c,            C,      copyright                 ] }; // c C ©
+    key <AC08> { [               s,            S,         ssharp                 ] }; // s S ß
+    key <AC09> { [               r,            R,     registered                 ] }; // r R ®
+    key <AC11> { [               m,            M,         macron,      masculine ] }; // m M ̄ º
+
+    key <AB02> { [               y,            Y,      braceleft                 ] }; // y Y {
+    key <AB03> { [               x,            X,     braceright                 ] }; // x X }
+    key <AB04> { [          period,        colon                                 ] }; // . :
+    key <AB05> { [               k,            K,     asciitilde                 ] }; // k, K, ~
+    key <AB09> { [               h,            H                                 ] }; // h H
+    key <AB10> { [               f,            F,              f,    ordfeminine ] }; // f F   ª
+
+    // Note : on a besoin de redéfinir les niveaux 3 et 4,
+    // donc nbsp(level2) ne suffit pas !
+    key <SPCE> { [           space,  nobreakspace,    underscore,   nobreakspace ] }; // ␣ (espace insécable) _ (espace insécable)
+};
+
+
+// Dernière version :
+
+partial alphanumeric_keys
+xkb_symbols "bepo" {
+
+    // reprend la version 1.0 :
+    include "fr(bepo_1_0)"
+
+    name[Group1]= "France - Bepo, ergonomic, Dvorak way";
 };
 
 partial alphanumeric_keys
@@ -529,12 +599,9 @@
 
     // Restricts the fr(bepo) layout to latin9 symbols
 
-    include "keypad(oss_latin9)"
-    include "fr(bepo)"
+    include "fr(bepo_1_0_latin9)"
 
     name[Group1]="France - Bepo, ergonomic, Dvorak way, latin-9 only";
-
-    // at now, no difference (oss keypad excepted), but will change !
 };
 
 
diff -Naur xkeyboard-config-1.3orig/symbols/inet xkeyboard-config-1.3bepo1.0/symbols/inet
--- xkeyboard-config-1.3orig/symbols/inet	2008-05-27 23:46:41.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/symbols/inet	2009-06-03 22:00:53.000000000 +0200
@@ -911,7 +911,7 @@
 partial alphanumeric_keys
 xkb_symbols "fscaa1667g" {
    include "inet(media_common)"
-   key <I32>   {       [ XF86Info              ]       };
+   key <I32>   {       [ XF86MyComputer        ]       };
    key <I6C>   {       [ XF86Mail              ]       };
 };
 
diff -Naur xkeyboard-config-1.3orig/symbols/symbols.dir xkeyboard-config-1.3bepo1.0/symbols/symbols.dir
--- xkeyboard-config-1.3orig/symbols/symbols.dir	2007-06-26 09:54:26.000000000 +0200
+++ xkeyboard-config-1.3bepo1.0/symbols/symbols.dir	2009-06-03 21:52:12.000000000 +0200
@@ -17,6 +17,8 @@
 --p----- a------- fr(latin9_Sundeadkeys)
 --p----- a------- fr(latin9_sundeadkeys)
 --p----- a------- fr(latin9_nodeadkeys)
+--p----- a------- fr(bepo_1_0)
+--p----- a------- fr(bepo_1_0_latin9)
 --p----- a------- fr(bepo)
 --p----- a------- fr(bepo_latin9)
 --p----- a------- fr(dvorak)
